src/Microsoft.Identity.Client/Internal/Cache/RefreshTokenCacheItem.cs

```diff
@@ -33,21 +34,80 @@ namespace Microsoft.Identity.Client.Internal.Cache
     [DataContract]
     internal class RefreshTokenCacheItem : BaseTokenCacheItem
     {
+
         public RefreshTokenCacheItem()
         {
         }
 
-        public RefreshTokenCacheItem(string authority, string clientId, TokenResponse response) : base(authority, clientId, response)
+        public RefreshTokenCacheItem(string environment, string clientId, TokenResponse response) : base(clientId)
         {
             RefreshToken = response.RefreshToken;
+            Environment = environment;
+            PopulateIdentifiers(response);
+
+            User = User.Create(DisplayableId, Name, IdentityProvider,
+                GetUserIdentifier());
         }
 
+        [DataMember(Name = "environment")]
+        public string Environment { get; set; }
+
+        [DataMember(Name = "uid")]
+        public string Uid { get; set; }
+
+        [DataMember(Name = "utid")]
+        public string Utid { get; set; }
```

Comment:
How is that going affect caching? If the properties are changed after the item has been put into the cache then the next time the cache key is calculated it will give a different value -> the existing cache entry won't be found.